lib/buffer.js

```diff
@@ -54,17 +54,19 @@ function alignPool() {
   }
 }
 
+const bufferDeprecationWarning =
+    deprecate(() => {}, 'Buffer constructor is deprecated. ' +
+      'Use Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.');
+
 /**
- * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
- * in the documentation and should not be used moving forward. Rather,
- * developers should use one of the three new factory APIs: Buffer.from(),
- * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
- * is no hard deprecation because of the extent to which the Buffer constructor
- * is used in the ecosystem currently -- a hard deprecation would introduce too
- * much breakage at this time. It's not likely that the Buffer constructors
- * would ever actually be removed.
+ * The Buffer() construtor is deprecated ... that is, it should not be used
+ * moving forward. Rather, developers should use one of the three new factory
+ * APIs: Buffer.from(), Buffer.allocUnsafe() or Buffer.alloc() based on their
+ * specific needs. It's not likely that the Buffer constructors would ever
+ * actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  bufferDeprecationWarning();
```

Comment:
@seishun 
> As far as I know, implementing it with the same behavior in `class Buffer` is quite complicated, if possible at all.

I was wrong on this. All we'd need to do is something like:
```js
class Buffer extends Uint8Array {
  constructor(string, encoding)  {
    const ab = binding.newArrayBufferFromString(string, encoding);
    super(ab);
  }
}
```

@addaleax There have been a couple suggestions of exporting a Buffer class that can be inherited, but it always wraps back around that it should be able to extend Buffer directly. For cleanliness of API I tend to agree, even if that involves breaking an egg or two.